src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/BuildsController.cs

```diff
@@ -143,6 +143,34 @@ public async Task<IActionResult> GetBuildGraph(int id)
             return Ok(new Models.Build(build));
         }
 
+        [HttpPatch("{buildId}")]
+        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(Build), Description = "Update a build with new information.")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> Update(int buildId, [FromBody, Required] BuildUpdate buildUpdate)
+        {
+            Data.Models.Build build = await _context.Builds.Where(b => b.Id == buildId).FirstOrDefaultAsync();
+
+            if (build == null)
+            {
+                return NotFound();
+            }
+
+            bool doUpdate = false;
+            if (buildUpdate.Released.HasValue)
+            {
+                build.Released = buildUpdate.Released.Value;
```

Comment:
Would make things more efficient if we checked that the old and new value are differents?

Reply:
Added a conditional there.